src/material/datepicker/calendar-body.scss

```diff
@@ -31,13 +32,24 @@ $calendar-range-end-body-cell-size:
   padding-right: $calendar-body-label-side-padding;
 }
 
-.mat-calendar-body-cell {
+.mat-calendar-body-cell-container {
+  display: table-cell;
   position: relative;
+  padding: 0;
   height: 0;
   line-height: 0;
+}
+
+.mat-calendar-body-cell {
+  @include button-common.reset();
+  position: absolute;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
```

Comment:
Well, that was the only way I could find to get that layout to be correct after spending an hour wrangling the CSS. It seemed liked the easiest way to me. We're already using absolute position to layout the other content inside the cell like the`.mat-calendar-body-cell-content` and the preview, so it seemed like a reasonable approach to me.  I'm open to suggestions :)

Here's what is looks like without the `position: absolute`
<img src="https://user-images.githubusercontent.com/7720245/148582480-c8334626-cfa8-49dd-8c61-c2718a9e7077.png" width=400/>

Reply:
@crisbeto the container being `display: table-cell` does seem to make this annoying (otherwise flexbox would solve it). I'm rusty here- is this really the best way to make this element 100% width/height?

Reply:
I think that this is our only option. The problem is that we set `padding-top`, `padding-bottom` and `width` on the table cell which distorts the content.